Mathlib/Data/Nat/ModEq.lean

```diff
@@ -0,0 +1,532 @@
+/-
+Copyright (c) 2017 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro
+
+! This file was ported from Lean 3 source module data.nat.modeq
+! leanprover-community/mathlib commit 09597669f02422ed388036273d8848119699c22f
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Data.Int.GCD
+import Mathlib.Data.Int.Order.Lemmas
+
+/-!
+# Congruences modulo a natural number
+
+This file defines the equivalence relation `a ≡ b [MOD n]` on the natural numbers,
+and proves basic properties about it such as the Chinese Remainder Theorem
+`ModEq_and_ModEq_iff_ModEq_mul`.
+
+## Notations
+
+`a ≡ b [MOD n]` is notation for `nat.ModEq n a b`, which is defined to mean `a % n = b % n`.
+
+## Tags
+
+ModEq, congruence, mod, MOD, modulo
+-/
+
+
+namespace Nat
+
+/-- Modular equality. `n.ModEq a b`, or `a ≡ b [MOD n]`, means that `a - b` is a multiple of `n`. -/
+def ModEq (n a b : ℕ) :=
+  a % n = b % n
+#align nat.ModEq Nat.ModEq
+
+@[inherit_doc]
+notation:50 a " ≡ " b " [MOD " n "]" => ModEq n a b
+
+variable {m n a b c d : ℕ}
+
+-- Porting note: This instance should be derivable automatically
+instance : Decidable (ModEq n a b) :=
+  if h : a % n = b % n then
+    isTrue h
+  else
+    isFalse h
+
+namespace ModEq
+
+@[refl]
+protected theorem refl (a : ℕ) : a ≡ a [MOD n] := rfl
+#align nat.ModEq.refl Nat.ModEq.refl
+
+protected theorem rfl : a ≡ a [MOD n] :=
+  ModEq.refl _
+#align nat.ModEq.rfl Nat.ModEq.rfl
+
+instance : IsRefl _ (ModEq n) :=
+  ⟨ModEq.refl⟩
+
+@[symm]
+protected theorem symm : a ≡ b [MOD n] → b ≡ a [MOD n] :=
+  Eq.symm
+#align nat.ModEq.symm Nat.ModEq.symm
+
+@[trans]
+protected theorem trans : a ≡ b [MOD n] → b ≡ c [MOD n] → a ≡ c [MOD n] :=
+  Eq.trans
+#align nat.ModEq.trans Nat.ModEq.trans
+
+instance : Trans (ModEq n) (ModEq n) (ModEq n) where
+  trans := Nat.ModEq.trans
+
+protected theorem comm : a ≡ b [MOD n] ↔ b ≡ a [MOD n] :=
+  ⟨ModEq.symm, ModEq.symm⟩
+#align nat.ModEq.comm Nat.ModEq.comm
+
+end ModEq
+
+theorem ModEq_zero_iff_dvd : a ≡ 0 [MOD n] ↔ n ∣ a := by rw [ModEq, zero_mod, dvd_iff_mod_eq_zero]
```

Comment:
Just a drive-by comment because I'm not an expert mathlib4 reviewer yet.  Here and in similar cases, I think the name should be `modEq_zero_iff_dvd`, according to [naming convention](https://github.com/leanprover-community/mathlib4/wiki/Porting-wiki#naming-convention) rule 5:
> When something named with `UpperCamelCase` is part of something named with `snake_case`, it is referenced in `lowerCamelCase`.

One could even argue that it should be `modeq_zero_iff_dvd` here, according to rule 6:
> Acronyms like LE are written upper-/lowercase as a group, depending on what the first character would be.

Reply:
I think `modEq` is the way to go here since we picked `ModEq` over `Modeq` for the prop.

Reply:
I think I got all fixed, at least both my eyes and grep can't seem to find an exception but probably someone should take a look again just in case.